rdt-app/src/main/java/io/ona/rdt_app/fragment/RDTJsonFormFragment.java

```diff
@@ -58,13 +64,34 @@ public void onClick(View v) {
                 Object isSubmit = v.getTag(R.id.submit);
                 if (isSubmit != null && Boolean.valueOf(isSubmit.toString())) {
                     save(false);
+                    checkIfContinuingToRdt();
+                    // Check if we need to continue to RDT
                 } else {
                     next();
                 }
             }
         });
     }
 
+    public void checkIfContinuingToRdt() {
+        String step1 = "step1";
+        if (step1.equals("step"+currentStep)) {
```

Comment:
Replace this with a check for encounter type.

Since we know the registration encounter type just has one step.